### ISSUE TITLE
Launching app on F-Droid

### ISSUE THREAD

**Issue by codedsun** [fossasia/phimpme-android#1181]
**Actual Behaviour**
The app has to be launched on F-Droid

**Expected Behaviour**

**Screenshots of the issue**

**Would you like to work on the issue?**

Yes ! @anantprsd5

**Comment by codedsun** [comment]
@anantprsd5 @mohitmanuja @pa1pal  This has to be done ! I will be needing help as @pa1pal is offline since days.

**Comment by Kristinita** [comment]
+1

**Comment by abishekvashok** [comment]
@codedsun any help i can deliver?

**Comment by codedsun** [comment]
@abishekvashok Yes, I had a vagrant error in the linux, I was involved in this process.I contacted their IRC Channel etc but no help was recieved. It would be great if we could take this issue together and solve this

**Comment by abishekvashok** [comment]
What priority software do we use? We don't right?

**Comment by realmain** [comment]
[Here](https://gitlab.com/fdroid/rfp) is where you submit a package for F-Droid btw.

**Comment by akavel** [comment]
"Closed solved"? Or "closed wontfix"? Or what? Is it there on F-Droid yet? [I don't seem to see it on the list](https://f-droid.org/en/packages/33/)

**Comment by realmain** [comment]
He closed this and opened it under #2018 for some reason

**Comment by sauravvishal8797** [comment]
@realmain Actually we had 2-3 issues regarding this matter, so I thought of opening a fresh one mentioning all the requirements for opening the merge request for Fdroid inclusion. Nevertheless, I'd close the #2018 and reopen this. And regarding the Fdroid inclusion, it is certainly at the top of our priority list. And at the moment we need to make some changes to our master branch and we're working on it, and once the changes are successfully implemented we're good to go and I'll be opening a merge request to Fdroid's repo for inclusion. Hopefully in a day or two we'd resolve the master branch issue and open the merge request.

**Comment by moreystraus** [comment]
If you release a new apk you could list it on the [izzy](
https://apt.izzysoft.de/fdroid/index.php) repo straight away.

**Comment by Robert321245** [comment]
Guys
Keep in mind that publish the app in the F-Droid store would give lot of visibility to the app in the Free and Open Source community and, in consequence, would bring new developers to the project.

**Comment by mariobehling** [comment]
@Robert321245 Could you help to make it happen bringing this app to Fdroid, please?

**Comment by iamareebjamal** [comment]
Requirements for F-Droid:

- [ ] Replace all hardcoded jars and native libraries. F-Droid needs all included libraries to be open source
- [ ] Split APK into Play Store and F-Droid builds where only Play Store APK has google-services.json and firebase/non open source libraries

**Comment by iamareebjamal** [comment]
@yashk2000 First part is done. Now please focus on second part

**Comment by yashk2000** [comment]
On it.

**Comment by iamareebjamal** [comment]
You'll have to see what other apps using OpenCV do and add a download and moving step in between

**Comment by iamareebjamal** [comment]
@yashk2000 -  
we cannot add the app directly to fdroid? OpenCV is open source.

@iamareebjamal -
```
git clone https://github.com/fossasia/phimpme-android
cd phimpme-android
./gradlew assembleRelease
```

Run these commands and tell me the output. This is what fdroid will run. If this runs without any error, then yes, you can add the app directly in fdroid

@yashk2000 -
we need to add the other files, `jniLibs`, `staticlibs`, `3rdparty` and `jni/include` too.

@iamareebjamal - 
> You'll have to see what other apps using OpenCV do and add a download and moving step in between

Exactly what I said before

@yashk2000 -
There used to be an app called OpenCV manager, but it is no longer there. I'll hunt for more ways to use OpenCV today afternoon. 
I'm sorry for the delay because of my college periodical exams this week. I'll compensate for it in the coming week.

**Comment by yashk2000** [comment]
@iamareebjamal I found only two apps with openCV on fdroid. One is Open Note Scanner. It provides the openCV manager, but it's using an older version of openCV, 3.x. The other app, Form-N-Fun using openCV also claims to provide the openCV manager, and it uses openCV4, but I was unable to download the app as it was not compatible with my device.

Please give some advice on how to proceed.

**Comment by iamareebjamal** [comment]
Use this https://github.com/iamareebjamal/opencv-android

**Comment by yashk2000** [comment]
@iamareebjamal you created an aar itself. Thanks :)

**Comment by yashk2000** [comment]
@iamareebjamal we still need to copy `staticlibs`, `jni/include`, `3rdParty` and `libs` from the openCV sdk.

**Comment by iamareebjamal** [comment]
Why?

**Comment by yashk2000** [comment]
The apk built successfully at that time because I still had those folders in the app/src/main directory. 
I deleted my local repository and recloned it. Then on building, it showed an error because of missing files.

**Comment by yashk2000** [comment]
I think it is because of the jni folder. It is the same as the jni in openCV but it has a few other files for applying the filters and effects. This contains the openCV.mk file which points to the staticlibs, 3rdpart and libs folders.

**Comment by iamareebjamal** [comment]
The app is building fine on Travis. List the errors which you are encountering?

**Comment by iamareebjamal** [comment]
> It is the same as the jni in openCV but it has a few other files for applying the filters and effects. 

Which other files are you talking about exactly?

> This contains the openCV.mk file which points to the staticlibs, 3rdpart and libs folders.

Which is also present in the aar file

**Comment by iamareebjamal** [comment]
> It is the same as the jni in openCV but it has a few other files for applying the filters and effects. 

Which other files are you talking about exactly?

> This contains the openCV.mk file which points to the staticlibs, 3rdpart and libs folders.

Which is also present in the aar file

**Comment by iamareebjamal** [comment]
When we removed JNI and other files from the repo, no error happened. Then why is it happening now?

**Comment by yashk2000** [comment]
It is building fine on travis because travis still has the openCV download script and copies the needed files.

**Comment by yashk2000** [comment]
Jni is not removed. It is there.

**Comment by iamareebjamal** [comment]
You should have disabled the script when migrating to aar

**Comment by iamareebjamal** [comment]
So I analyzed the way things were working and it is extremely confusing. The opencv folder is used as a module, but it did not contain any jni code. All its native files are in phimpme package because they are needed to compile certain custom filter code. Don't understand how that was working, but I am reverting the AAR file. Please explain below how and why things were working

For now, adapt the travis script to be used with F-Droid build process

**Comment by yashk2000** [comment]
I myself am not very clear about how the openCV files present in the phimpme package work directly from there. @abishekvashok could you please help with this part?

How do I adapt the travis script to be used with F-Droid build process? We need the script to download the openCV sdk and copy the needed files to phimpme package.

**Comment by iamareebjamal** [comment]
Remove the sudo apt commands and make a single script and test with fdroid

**Comment by yashk2000** [comment]
I'll do it by today evening.

**Comment by yashk2000** [comment]
@iamareebjamal I have been working on the script. But I am facing some errors. 
[This](https://travis-ci.org/yashk2000/phimpme-android/builds/564368602) is one of the things I tried. I also tried to use wget <link> -P $HOME/openCV. But then when I try to move the $HOME/openCV-android-sdk to $HOME/openCV/opencv, it shows no such file or directory. Could you please help.

**Comment by iamareebjamal** [comment]
Don't use $HOME. Just use relative paths

**Comment by iamareebjamal** [comment]
You already have a script of travis. You just have to rearrange it and move some commands

**Comment by yashk2000** [comment]
> Don't use $HOME. Just use relative paths

Ok. I'll try that.

> You already have a script of travis. You just have to rearrange it and move some commands

Yes. I am just facing this path problem.

**Comment by iamareebjamal** [comment]
@yashk2000 Please create a merge request on Fdroid repo with metadata YML with the setup_opencv script to include the app

**Comment by yashk2000** [comment]
@iamareebjamal in the org.fossasia.phimpme.yml, how do I put the setup_opencv script so that it is also run?

**Comment by iamareebjamal** [comment]
Read the docs https://f-droid.org/en/docs/Build_Metadata_Reference/#Build

**Comment by yashk2000** [comment]
@iamareebjamal so I can execute the script using:
`build: ./scripts/opencv_setup.sh`

**Comment by iamareebjamal** [comment]
No, that is build step to build the apk

**Comment by iamareebjamal** [comment]
I don't think you even opened the link and read about different options.

**Comment by yashk2000** [comment]
I thought that I had to read the build option only because in the link you gave, it's #Build at the end. Sorry. I'll go through all the options.

**Comment by iamareebjamal** [comment]
You have to read the build **section** only. That's why I gave you the link

**Comment by yashk2000** [comment]
Whether I open the link with or without the #Build extension, it is always leading me to the landing page of the Build Metadata Reference.

I went through the options. Do I have to use the `sudo` command?

**Comment by iamareebjamal** [comment]
No

**Comment by iamareebjamal** [comment]
https://f-droid.org/en/docs/Build_Metadata_Reference/#Builds

**Comment by yashk2000** [comment]
@iamareebjamal the commands I found for running bash commands are `sudo`, `prebuild`, `build`, and `init`.

**Comment by iamareebjamal** [comment]
So which one do you think should be used?

**Comment by yashk2000** [comment]
`prebuild`? We need to copy the folders before the build starts.

**Comment by iamareebjamal** [comment]
Yes

**Comment by yashk2000** [comment]
@iamareebjamal so I store the opencv_setup in the scrlib directory and then use it by `prebuild: $$opencv_setup$$` ?

**Comment by iamareebjamal** [comment]
You already have the script in scripts folder

**Comment by yashk2000** [comment]
@iamareebjamal https://gitlab.com/fdroid/fdroiddata/merge_requests/5164

**Comment by iamareebjamal** [comment]
Add this change in prebuild as well https://gitlab.com/fdroid/fdroiddata/merge_requests/5053/diffs

**Comment by yashk2000** [comment]
@iamareebjamal is it fine now? 
https://gitlab.com/fdroid/fdroiddata/merge_requests/5164

**Comment by ByteHamster** [comment]
If you remove the description from that file, F-Droid will take the description from this repository's fastlane folder. That allows to change and translate the description more easily.

**Comment by iamareebjamal** [comment]
> is it fine now?

No, read the documentation again on how you add multiple commands in prebuild.

https://f-droid.org/en/docs/Build_Metadata_Reference/#Builds

**Comment by yashk2000** [comment]
> > is it fine now?
> 
> No, read the documentation again on how you add multiple commands in prebuild.
> 
> https://f-droid.org/en/docs/Build_Metadata_Reference/#Builds

Yes I tried to do `scripts/setup_opencv.sh && <the other command>`.

But the pipeline showed a lint error and showed that it had to be written like it is now.

**Comment by iamareebjamal** [comment]
OK

**Comment by yashk2000** [comment]
https://gitlab.com/fdroid/fdroiddata/merge_requests/5164#note_200055659

@iamareebjamal should I remove the maven dependencies? I removed them and built the app. There wasn't any problem.

**Comment by iamareebjamal** [comment]
Yes

**Comment by yashk2000** [comment]
@iamareebjamal https://github.com/fossasia/phimpme-android/pull/2851

**Comment by ammgws** [comment]
What's the status on this? Anything left to do?

**Comment by iamareebjamal** [comment]
Yes, a lot of things remaining https://gitlab.com/fdroid/fdroiddata/-/merge_requests/5877

**Comment by asaikarthikeya** [comment]
@iamareebjamal Hello sir , I am willing to contribute can you please guide me.

**Comment by meximan** [comment]
Any updates on the f droid merge?

**Comment by ghost** [comment]
> Any updates on the f droid merge?

It looks like this app might have been abandoned. It hasn't had a commit since 2020 Oct 12th. All the other commits were bot commits to bump up dependency versions. Last release was 2020 Aug 15th.

**Comment by meximan** [comment]
That's unfortunate

**Comment by goyalyashpal** [comment]
2022 March 10
